src/workerd/api/util-test.c++

```diff
@@ -53,5 +59,120 @@ KJ_TEST("redactUrl can detect base64 ids") {
   expectUnredacted("https://domain/IThinkIShallNeverSee0/x"_kj);
 }
 
+KJ_TEST("readContentTypeParameter can fetch boundary parameter") {
+
+  // normal
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // multiple params
+  expectContentTypeParameter(
+    "multipart/form-data; charset=utf-8; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+ // param name inside value of other param
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // ensure param is not found
+  KJ_ASSERT(readContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=\"__boundary__\""_kj,
+    "boundary1"_kj
+  ) == nullptr);
+
+  // no quotes
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=__boundary__"_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // attribute names are case-insensitive, but values are not
+  expectContentTypeParameter(
+    "multipart/form-data; charset=\"boundary=;\"; boundary=__Boundary__"_kj,
+    "Boundary"_kj,
+    "__Boundary__"_kj
+  );
+
+  // different order
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=\"__boundary__\"; charset=utf-8"_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // bogus parameter
+  expectContentTypeParameter(
+    "multipart/form-data; foo=123; boundary=\"__boundary__\""_kj,
+    "boundary"_kj,
+    "__boundary__"_kj
+  );
+
+  // quoted-string
+  expectContentTypeParameter(
+    R"(multipart/form-data; foo="\"boundary=bar\""; boundary="realboundary")",
+    "boundary"_kj,
+    "realboundary"_kj
+  );
+
+  // handle non-closing quotes
+  KJ_ASSERT(readContentTypeParameter(
+    R"(multipart/form-data; charset="boundary=;\"; boundary="__boundary__")",
+    "boundary"_kj
+  ) == nullptr);
+
+  // handle non-closing quotes on wanted param
+  KJ_ASSERT(readContentTypeParameter(
+    R"(multipart/form-data; charset="boundary=;"; boundary="__boundary__\")",
+    "boundary"_kj
+  ) == nullptr);
+
+  // handle incorrect quotes
+  KJ_ASSERT(readContentTypeParameter(
+    R"(multipart/form-data; charset=\"boundary=;\"; boundary=\"__boundary__\")",
+    "boundary"_kj
+  ) == nullptr);
+
+  // spurious whitespace before ;
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=asdf ;foo=bar"_kj,
+    "boundary"_kj,
+    "asdf"_kj
+  );
+
+  // spurious whitespace before ; with quotes
+  expectContentTypeParameter(
+    "multipart/form-data; boundary=\"asdf\" ;foo=bar"_kj,
+    "boundary"_kj,
+    "asdf"_kj
+  );
+
+  // all whitespace
+  KJ_ASSERT(readContentTypeParameter(
+    "multipart/form-data; boundary= ;foo=bar"_kj,
```

Comment:
Other variations to check in the test are:

* `multipart/form-data; boundary= a`
* `multipart/form; boundary=" a"` 

In both cases the `boundary` parameter value should be `" a"` (space followed by `a`)

RFC2046 defines the `boundary` construct as `boundary := 0*69<bchars> bcharsnospace`, which means it can *start* with a space but does not end with a space.

Reply:
Added those 2 testcases.